.github/issue-responses/common-issues-guide.md

```diff
@@ -0,0 +1,1208 @@
+4. [Broker Integration Issues](#4-broker-integration-issues)
+5. [Build and Dependency Issues](#5-build-and-dependency-issues)
+6. [Runtime Crashes](#6-runtime-crashes)
+7. [Single vs Multiple Account Mode Issues](#7-single-vs-multiple-account-mode-issues)
+8. [Silent Token Refresh Issues](#8-silent-token-refresh-issues)
+9. [Network and Cloud Discovery Issues](#9-network-and-cloud-discovery-issues)
+10. [WebView and Browser Issues](#10-webview-and-browser-issues)
+11. [Azure AD B2C Specific Issues](#11-azure-ad-b2c-specific-issues)
+
+---
+
+## 1. Configuration Issues
+
+### 1.1 Redirect URI Encoding Mismatch
+
+**Symptoms:**
+- "Redirect URI mismatch" error during authentication
+- Authentication fails with no clear error message
+- `MsalClientException` with error code `redirect_uri_validation_error`
+
+**Root Cause:**
+The signature hash in `auth_config.json` must be URL encoded, while the same hash in `AndroidManifest.xml` must NOT be URL encoded.
+
+**Solution:**
+
+```xml
+<!-- AndroidManifest.xml - NOT URL encoded -->
+<data
+    android:scheme="msauth" 
+    android:host="your.package.name"
+    android:path="/ABcDeFg+okk=" />
+```
+
+```json
+// auth_config.json - MUST be URL encoded
+{
+    "redirect_uri": "msauth://your.package.name/ABcDeFg%2Bokk%3D"
+}
+```
+
+**URL Encoding Reference:**
+| Character | URL Encoded |
+|-----------|-------------|
+| `+` | `%2B` |
+| `=` | `%3D` |
+| `/` | `%2F` |
+| `*` | `%2A` |
+
+**Related Documentation:** [Configuration Guide](../../auth_config.template.json)
+
+---
+
+### 1.2 Missing or Incorrect Client ID
+
+**Symptoms:**
+- `MsalClientException: client_id` error
+- PCA initialization fails
+- "Application not found" error from Azure AD
+
+**Root Cause:**
+The `client_id` in `auth_config.json` doesn't match the Azure App Registration.
+
+**Solution:**
+1. Verify the Client ID in Azure Portal → App Registrations → Your App → Overview
+2. Update `auth_config.json`:
+```json
+{
+    "client_id": "YOUR_CORRECT_CLIENT_ID_FROM_AZURE_PORTAL"
+}
+```
+
+---
+
+### 1.3 Missing BrowserTabActivity in AndroidManifest.xml
+
+**Symptoms:**
+- Authentication starts but never completes
+- App doesn't receive the authentication response
+- Screen goes blank after entering credentials
+
+**Solution:**
+Add the BrowserTabActivity to your AndroidManifest.xml:
+
+```xml
+<activity
+    android:name="com.microsoft.identity.client.BrowserTabActivity"
+    android:exported="true">
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW" />
+        <category android:name="android.intent.category.DEFAULT" />
+        <category android:name="android.intent.category.BROWSABLE" />
+        <data
+            android:scheme="msauth" 
+            android:host="YOUR_PACKAGE_NAME"
+            android:path="/YOUR_SIGNATURE_HASH" />
+    </intent-filter>
+</activity>
+```
+
+---
+
+## 2. Authentication Errors
+
+### 2.1 AADSTS Error Codes
+
+**Common AADSTS Errors and Solutions:**
```

Comment:
>*Common AADSTS Errors and Solutions:** [](http://example.com/codeflow?start=1&length=38)

Additionally can link to [https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes](https://learn.microsoft.com/en-us/entra/identity-platform/reference-error-codes)